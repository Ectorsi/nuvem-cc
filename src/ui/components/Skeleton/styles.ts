import styled, { css } from "styled-components";
import { SkeletonProps } from "./Skeleton";

const skeletonModifiers = {
    small: () => `
        width: 20px;
        height: 10px;
    `,
    large: () => `
        width: 100px;
        height: 100px;
    `,
};

export const SkeletonContainer = styled.div<SkeletonProps>`
    ${({ size }) => css`
        background-image: linear-gradient(90deg, #dddddd 0%, #eeeeee 50%, #dddddd 100%);
        background-size: 200px 100%;
        animation: loading 0.8s infinite linear;
        @keyframes loading {
            0% {
                background-position: -200px 0;
            },
            100% {
                background-position: calc(200px + 100%) 0;
            }
        };
        border-radius: 0.5rem;
        width: 40%;
        height: 30px;
        align-self: center;
        ${!!size && skeletonModifiers[size]};
    `}
`;