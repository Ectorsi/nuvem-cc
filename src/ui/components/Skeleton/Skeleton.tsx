
import * as S from './styles';

export type SkeletonProps = {
    size?: 'small' | 'large';
};

const Skeleton = ({ size }: SkeletonProps) => (
    <S.SkeletonContainer data-testid="skeleton" size={size} />
);

export default Skeleton;