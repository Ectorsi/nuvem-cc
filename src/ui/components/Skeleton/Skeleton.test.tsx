import { render, screen } from '@testing-library/react';

import Skeleton, { SkeletonProps } from './Skeleton';

const SkeletonTestProps: SkeletonProps = {
    size: undefined,
};

describe('Skeleton', () => {
    it('should be able to match snapshot', () => {
        render(<Skeleton {...SkeletonTestProps} />);
        expect(screen.getByTestId('skeleton')).toMatchSnapshot();
    });
    it('should render correctly', () => {
        render(<Skeleton {...SkeletonTestProps} />);
        expect(screen.getByTestId('skeleton')).toBeInTheDocument();
    });
    it('should render small size', () => {
        render(<Skeleton size='small' />);
        expect(screen.getByTestId('skeleton')).toHaveStyle('width: 20px');
    });
    it('should render large size', () => {
        render(<Skeleton size='large' />);
        expect(screen.getByTestId('skeleton')).toHaveStyle('width: 100px');
    });
});