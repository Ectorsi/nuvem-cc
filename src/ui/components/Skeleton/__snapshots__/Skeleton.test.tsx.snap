// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton should be able to match snapshot 1`] = `
<div
  class="sc-beqWaB hcpvpR"
  data-testid="skeleton"
/>
`;
